File: Arrays/Two Dimensional Arrays/rotateMatrix.py
n = int(input())

matrix = []

# taking array input
for i in range(n):
    arr = []
    for j in range(n):
        arr.append(int(input()))
    matrix.append(arr)


def display(arr, n):
    for i in range(n):
        for j in range(n):
            print(arr[i][j], end=" ")
        print()


def rotate(arr, n):
    for i in range(n - 1):
        startCol = 0
        endCol = n - 1
        while (startCol < endCol):
            arr[i][endCol], arr[i][startCol] = arr[i][startCol],arr[i][endCol]

            startCol += 1
            endCol -= 1

    # for transpose
    for i in range(n - 1):
        for j in range(n - 1):
            if (i < j):
                arr[i][j],arr[j][i] = arr[j][i], arr[i][j]


display(matrix, n)
rotate(matrix, n)
display(matrix, n)

File: Arrays/Two Dimensional Arrays/rotateMatrix.cpp
#include <bits/stdc++.h>
using namespace std;

void display(int arr[][100], int n){
    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            cout << arr[i][j] << " ";
        }
        cout << endl;
    }
}

void rotate(int arr[][100], int  n){

    //reverse each row
    for (int i = 0; i < n; i++){
        int startCol = 0;
        int endCol = n - 1;
        while (startCol < endCol){
            swap(arr[i][startCol], arr[i][endCol]);
            startCol += 1;
            endCol -= 1;
        }
    }

    //transpose
    for (int i = 0; i < n;i++){
        for (int j = 0; j < n; j++){
            if (i < j){
            swap(arr[i][j], arr[j][i]);
            }
        }
    }
}

int main(){
    
    int arr[100][100];
    int n;
    cin >> n;

    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            cin >> arr[i][j];
        }
    }


    rotate(arr, n);
    display(arr, n);


    return 0;
}
